rackcity/utils/asset_utils.py

```diff
@@ -507,17 +507,20 @@ def save_all_field_data_cp(data, asset, change_plan, create_asset_cp):
             setattr(asset, field, value)
     chassis = None
     if data["chassis"]:
-        try:
+        if Asset.objects.filter(id=data["chassis"]).exists():
             chassis_live = Asset.objects.get(id=data["chassis"])
-        except ObjectDoesNotExist:
-            return (None, "No existing chassis with id=" + data["chassis"])
-
-        if not AssetCP.objects.filter(id=data["chassis"]).exists():
-            chassis = add_chassis_to_cp(
-                chassis_live, change_plan, ignore_blade_id=asset.id
-            )
+            if not AssetCP.objects.filter(id=data["chassis"]).exists():
+                chassis = add_chassis_to_cp(
+                    chassis_live, change_plan, ignore_blade_id=asset.id
+                )
```

Comment:
noticed this big when i was debugging chassis permissions bug -> if that chassis doesn't exist in Asset, we need to check AssetCP before returning error

Reply:
what if the chassis were created on a CP as well, so there's no Asset for it? confused about this codepath

Reply:
oh hm good point. im confused about this code path too. rewriting it